coalahtml/tree.py

```diff
@@ -0,0 +1,84 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+    """
+    Node class
+    """
+    count = 1
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+        self._id = Node.count
+        Node.count += 1
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(os.path.commonprefix(file_list))
+        self.root.abspath = self.root.path
+        self.fl_split = Tree._split_paths(file_list)
+        self._build(self.fl_split)
+
+    @staticmethod
+    def _split_paths(path_list):
+        return [path.split('/') for path in path_list]
```

Comment:
It's only that according to the python docs, using split is preferred method. I'm not aware of all the nuances.